client/app/competitor/page.tsx

```diff
@@ -103,14 +111,14 @@ const EditorButtons = ({ isPaused }: EditorButtons) => {
                     </Button>
                 </Tooltip>
                 <span className="ml-auto">
-                    <Select>
-                        <SelectTrigger className="w-56" defaultValue={currQuestion?.languages?.[0]}>
+                    <Select value={selectedLanguage} onValueChange={setSelectedLanguage}>
+                        <SelectTrigger className="w-56">
                             <SelectValue placeholder="Programming Language" />
                         </SelectTrigger>
                         <SelectContent>
                             {currQuestion?.languages?.map((l) => (
-                                <SelectItem key={l} value={l}>
-                                    {l}
+                                <SelectItem key={l.language} value={l.syntax}>
+                                    {l.language}
```

Comment:
This will break if there are two languages using the same syntax.  The value here should be the name of the language.  i.e., Python, Java